crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -209,6 +327,106 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
+    MethodRequired,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct DeutschebankThreeDSInitializeResponseChallengeRequired {
+    acs_url: String,
+    creq: String,
+}
+
+impl
+    TryFrom<
+        ResponseRouterData<
+            Authorize,
+            DeutschebankThreeDSInitializeResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
+    > for RouterData<Authorize, PaymentsAuthorizeData, PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<
+            Authorize,
+            DeutschebankThreeDSInitializeResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        match item.response.outcome {
+            DeutschebankThreeDSInitializeResponseOutcome::Processed => Ok(Self {
+                status: match item.data.request.is_auto_capture()? {
+                    true => common_enums::AttemptStatus::Charged,
+                    false => common_enums::AttemptStatus::Authorized,
+                },
+                response: Ok(PaymentsResponseData::TransactionResponse {
+                    resource_id: ResponseId::NoResponseId,
+                    redirection_data: Box::new(None),
+                    mandate_reference: Box::new(None),
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                    incremental_authorization_allowed: None,
+                    charge_id: None,
+                }),
+                ..item.data
+            }),
+            DeutschebankThreeDSInitializeResponseOutcome::ChallengeRequired => {
+                match item.response.challenge_required {
+                    Some(challenge) => Ok(Self {
+                        status: common_enums::AttemptStatus::AuthenticationPending,
+                        response: Ok(PaymentsResponseData::TransactionResponse {
+                            resource_id: ResponseId::NoResponseId,
+                            redirection_data: Box::new(Some(
+                                RedirectForm::DeutschebankThreeDSChallengeFlow {
+                                    acs_url: challenge.acs_url,
+                                    creq: challenge.creq,
+                                },
+                            )),
+                            mandate_reference: Box::new(None),
+                            connector_metadata: None,
+                            network_txn_id: None,
+                            connector_response_reference_id: None,
+                            incremental_authorization_allowed: None,
+                            charge_id: None,
+                        }),
+                        ..item.data
+                    }),
+                    None => Err(errors::ConnectorError::MissingRequiredField {
```

Comment:
We throw a `MissingRequiredField` error when we fail to receive a mandatory parameter from the user to construct a request. In this case, throw an `UnexpectedResponseError` instead